M2/Macaulay2/d/version.dd

```diff
@@ -104,6 +104,29 @@ header "#include <readline/readline.h>";
 header "#include <gdbm.h>";
 gdbmversion():constcharstar := Ccode(returns, "return gdbm_version");
 
+header "#include <dlfcn.h>";
+blasimplementation():constcharstar := Ccode(returns, "
```

Comment:
This seems kind of hacky. Why not just set the implementation type in configure/cmake and put it in a header file? e.g. cmake build prints `LAPACK_LIBRARIES` for the user:
```m2
## Library information
     Linear Algebra    = /usr/lib64/libflexiblas.so;-lm;-ldl
```

Reply:
I don't think this needs to be a sentence necessarily, or even match in the cmake/autotools builds. Either listing the library type, library files, or even just a keyword probably suffices.

Reply:
My rationale for having this at runtime was to account for the "update-alternatives" system in Debian/Ubuntu (and I think flexiblas does something similar) where you can change the BLAS implementation on the fly.  The M2 binary is linked to `libblas.so`, but it's just a symlink to whichever BLAS you want to run.  We call `realpath` so we can see what it's symlinked to.

Reply:
I think flexiblas and openblas are intentionally written to be hot-swappable, but this is definitely not always the case. e.g. MKL can't be swapped in at runtime. If you want you could add a top-level way to check this (e.g. set it in version.m2), but I'm still against making things so complicated because just calling `ldd` is way simpler than all of this.

Reply:
Fair enough -- I don't feel too strongly about this.  :)  I'll remove this commit soon.